source/linux/epoll_event_loop.c

```diff
@@ -286,11 +288,19 @@ static void s_stop_task(struct aws_task *task, void *args, enum aws_task_status
          */
         epoll_loop->should_continue = false;
     }
+    aws_atomic_store_ptr(&epoll_loop->stop_task_ptr, NULL);
```

Comment:
Should we reset the atomic BEFORE setting `epoll_loop->should_continue`?
At this point the task is out of the scheduler and it's safe for someone else to get it re-scheduled via a call to stop()

Reply:
:+1: